BinaryConverter/src/BinaryConverter.java

```diff
@@ -0,0 +1,366 @@
+		public static final int FLOAT_TYPE = 2;
+	
+	public static void main(String[] args) {
+		
+		int dataType = -1;
+		Scanner s = null;
+		boolean allGood = false;
+		try {
+			s = new Scanner( new File("testClass.txt"));
+			//System.out.println(It's open");
+			allGood = true;
+		} catch( Exception e ) {
+			System.out.println("It's not open. There is an issue opening your file. %s" + e);
+		}
+		
+		if(allGood) {
+			while(s != null && s.hasNextLine()) {
+				String nextNumIn = s.nextLine();
+				
+				if(itsNotANumber (nextNumIn)) {
+					//check for bad input
+					System.out.println(nextNumIn + " is bad input");
+					dataType = STRING_TYPE;
+				} else if(isADecimal(nextNumIn)) {
+					//check for a decimal
+					System.out.println(nextNumIn + " is a decimal");
+					dataType = FLOAT_TYPE;
+					binConvertDec(Double.parseDouble(nextNumIn));
+				} else {
+					//it's an integer
+					System.out.println(nextNumIn + " is an integer");
+					dataType = INT_TYPE;
+				}
+				
+				//now process the numbers....
+				if(dataType == INT_TYPE) {
+					String binInt = "";
+					try {
+						long x = Long.parseLong(nextNumIn);
+						//This is my conversion method...
+						binInt = binConvertInt(x);
+						System.out.println(binInt);
+						
+					} catch (Exception e) {
+						System.out.println(nextNumIn + " -- Number is too large");
+						binInt = "" + nextNumIn + " is too big for Long";
+					}
+					
+				} else if (dataType == FLOAT_TYPE) {
+					//create the 32 bit float method
+				} else {
+					//Strings??
+				}
+			}
+			s. close();
+		}
+	}
+	
+	public static boolean itsNotANumber(String nextNumIn) {
+		boolean itsNotANumber = false;
+		String nonNums = "abcdefghijklmnopqrstuvwxyz!@#$%^&*()_+= {[}]|\\:;\"'<,>?/";
+		
+		for(int l= 0; l < nonNums.length(); l++) {
+			if(nextNumIn.contains("" + nonNums.charAt(l))) {
+				itsNotANumber = true;
+			}
+		}
+		return itsNotANumber;
+	}
+	
+	public static boolean isADecimal(String nextNumIn) {
+		boolean itsADecimal = false;
+		if(nextNumIn.contains(".")) {
+			itsADecimal = true;
+		}
+		return itsADecimal;
+	}
+	
+	
+	
+	public static int findBits(double num) {
+		
+		if(num >= -128 && num <= 127) return 8;
+		if(num >=  -32768 && num <= 32767) return 16;
+		if(num >= -Math.pow(2, 31) && num <= Math.pow(2, 31) - 1) return 32;
+		if(num >= -Math.pow(2, 63) && num <= Math.pow(2, 63) - 1) return 64;
+		
+		return -1;
+	}
+		
+		
+		
+	public static String binConvertDec(double dNum) {
+		
+		String bitString = "";
+		double startNum = dNum;
+		
+		
+		// Now get negative bit
+		String negBit = "";
+		
+		if(dNum >= 0) {
+			negBit = "0";
+		} else {
+			negBit = "1";
+			dNum *= -1;
+		}
+		System.out.println("The negative bit is: " + negBit);
+		
+		
+		//separating int. and dec. part of input number
+		long intNum = (long)dNum;
+		double decNum = dNum - intNum;
+		
+		System.out.println(decNum);
+		
+		
+		//This could be your character array....
+				
```

Comment:
You can try separating the decimal and the integer part because you only need to write an extra part for the decimals. You can create a variable called "int integerPart" and call the binConvertInt() method (you already have a pretty good version of it). Then create a variable called "double decPart" and do mantissa on it myself.